### PR TITLE
Optimizations and Fixes

### PR DESCRIPTION
**Pull request by mehah**
1- Walk System Fix
2- Support Idle Animation
3- Monster Light Fix
4- Optimized Render System
5- Compile Warnings fix

### CODE REVIEW

**Comment by slavidodo** [comment]
Too much in one commit, I can't review 112 files all at once :)
I'd really appreciate if u addressed each of the issues on a single PR!

**Comment by mehah** [comment]
Sry, but most of the files are VS indentation.

**Comment by andersonfaaria** [comment]
> Too much in one PR, I can't review 112 files all at once :)
> I'd really appreciate if u addressed each of the issues on a single PR!

I'm sorry slavi, but separating in different PRs would reduce the file amount? I don't get it...
You just need to check one at time, after you finish you press the "Viewed" option.

**Comment by slavidodo** [comment]
You're fixing 5 issues, it makes sense if each issue is actually solved on its own PR.
And yes, less files means less time to spend reviewing the whole PR and you still need to revert these indentations. We need to keep our style!

**Comment by mehah** [comment]
@slavidodo 

indentation is correct, according to these rules at this link: https://github.com/edubart/otclient/wiki/OTClient-Coding-Style

it was actually more formatting and spaces were removed where they were not needed.

In short, cleanliness.

**Comment by iryont** [comment]
Indentation is correct, but @slavidodo meant style which is not correct, e.g.:

```c++
if (x) {
```

Should be changed back to:

```c++
if(x) {
```

The style should be respected as it was originally intended to be.

For anyone willing to review this PR it is very difficult to read as most lines are about changed style and it is certainly not friendly to focus on.

**Comment by mehah** [comment]
ok, no problem.

**Comment by lgrossi** [comment]
This repo is almost die, so many 'blocks' to an healthy addition, almost looks like you guys don't want it to evolve properly..

File: src/client/tile.h [hunk 1]
```diff
@@ -87,9 +87,9 @@ class Tile : public LuaObject
     const Position& getPosition() { return m_position; }
     int getDrawElevation() { return m_drawElevation; }
     std::vector<ItemPtr> getItems();
-    std::vector<CreaturePtr> getCreatures();
     std::vector<CreaturePtr> getWalkingCreatures() { return m_walkingCreatures; }
     std::vector<ThingPtr> getThings() { return m_things; }
+    std::vector<CreaturePtr> Tile::getCreatures() { return m_creatures; }
```
Comment (lgrossi) [inline src/client/tile.h:92]: remove "tile::"

**Comment by diath** [comment]
I'm closing this PR because there's no way this is getting merged, if you want to realistically have those changes reviewed and merged here then consider the following:
1) Revert the stylistic changes, even if there is an issue with consistency we can focus on reviewing and merging these features and improvements first, and resolving code consistency across the codebase later once we establish something that people agree on (honestly it would be for the best if we had clang format file for this that tried to replicate the original style as closely as possible). Changing the style of the codebase on a whim isn't the way to go.
2) Revert the changes that break compiling under GCC, I can't even test the changes at all!
3) Revert the changes that remove grouping parens from hundreds of expressions, removing these generates hundreds of warnings because they are here to help guard against common human errors.
4) Split each feature into a feature branch and submit the branches as separate PRs. This will help review and address issues with your code on a feature basis which not only makes it easier for everyone involved but also won't block the entire PR from getting merged if one features is fine but other is yet to be addressed. Plus when you make unrelated changes we won't have the PR randomly updating with things that are irrelevant.

**Comment by soul4soul** [comment]
@mehah diath's comments are valid and expected in nearly all OSS development. Please take his comments as education and not discouragement. I sincerely hope you open open PRs with the issues addressed.

**Comment by iryont** [comment]
Let's grab popcorn now and watch conspiracy theories about why this PR was closed and how @slavidodo is not merging useful PRs because he does not want OTClient to compete with his own game client. The very same people will later cry about good developers leaving OT community. 

This toxic and demanding attitude of some is precisely the reason why initial developers left.

**Comment by mehah** [comment]
> I'm closing this PR because there's no way this is getting merged, if you want to realistically have those changes reviewed and merged here then consider the following:
> 
> 1. Revert the stylistic changes, even if there is an issue with consistency we can focus on reviewing and merging these features and improvements first, and resolving code consistency across the codebase later once we establish something that people agree on (honestly it would be for the best if we had clang format file for this that tried to replicate the original style as closely as possible). Changing the style of the codebase on a whim isn't the way to go.
> 2. Revert the changes that break compiling under GCC, I can't even test the changes at all! I'd love to point out which but I honestly can't be bothered scrolling through 106 files (see the issue here?).
> 3. Revert the changes that remove grouping parens from hundreds of expressions, removing these generates hundreds of warnings because they are here to help guard against common human errors.
> 4. Split each feature into a feature branch and submit the branches as separate PRs. This will help review and address issues with your code on a feature basis which not only makes it easier for everyone involved but also won't block the entire PR from getting merged if one features is fine but other is yet to be addressed. Plus when you make unrelated changes we won't have the PR randomly updating with things that are irrelevant.

1 - This has already been done and you simply ignored it. https://github.com/edubart/otclient/pull/1093/commits/4c630b448aa1f267015e45dac2fed413346bc2e9

2/3 - You compiled the branch in which I spent a cleanup, it was to have compiled the master, in which PR was made.

4- I'm sorry, but unfortunately it will not be possible to open a PR separately for each change.

**Pull request closed without merging**